compiler/qsc_hir/src/hir.rs

```diff
@@ -174,6 +174,8 @@ pub struct ItemId {
     pub package: Option<PackageId>,
     /// The item ID.
     pub item: LocalItemId,
+    /// The item status.
+    pub status: ItemStatus,
```

Comment:
It's very strange to me that this is part of the `ItemId`. It's... not an ID. It's part of the item data. There are a _lot_ of places in the code that use `ItemId` as hashmap keys and now we're adding another field to hash for no good reason. Putting it in the `ItemId` also leaks this field into a bunch of places that don't need it, like the language service.

If this attr _has_ to be checked in the resolver, maybe it can just be another field in `resolve::Res::Item`? 

(Though you already know that my real preference is for this logic not to be in the resolver at all... the HIR `Item` will already have the attributes on it, it could be so easy just to make this check a pass. )

Reply:
I can try out what the code looks like by having it be a part of the `hir::Res::Item` and `resolve::Res::Item`. I think you are right that the latter would affect fewer places.

As for making it a pass, I think if we did that it would have to go in the HIR somewhere. While you are correct that the original item has the attributes on it, right now there are no passes that use the global table to look up resolved references from other parts of the package or dependent packages (note that `run_default_passes` doesn't take any global lookup tables, nor a reference to the compilation store). To have a pass check all resolutions by following the link and looking up the corresponding package from the store would be very different and upend the nice pattern we have today where passes only run on the package being compiled. Like type info, whether or not a resolution is pointing to a unimplemented item is information we have earlier in the compilation that we can put into the HIR itself to save later phases lookup.

All that said, right now I'm not using this info in later phases, so that's where your first suggestion of keeping this scoped to the resolver by using `resolve::Res::Item` fits with the current approach.

Reply:
I think we landed at a better place for this. If we do end up deciding to move this (and any upcoming logic) to a pass, we can reevaluate.